python/rikai/spark/sql/callback_service.py

```diff
@@ -47,29 +48,38 @@ class CallbackService:
 
     def __init__(self, spark: SparkSession):
         self.spark = spark
+        self.registry_map = {}
 
     def __repr__(self):
         return "PythonCbService"
 
-    def codegen(self, model, temporary: bool):
-        """Code generation for a Model.
+    def resolve(
+        self, registry_class: str, uri: str, name: str, options: JavaMap
+    ):
+        """Resolve a ML model.
 
         Parameters
         ----------
-        model : jvm Model class
-            The model to generate python code.
-        temporary : bool
-            Set true of the generated code will only be used once. Temporary
-            code generation will be used when we use `ML_PREDICT` directly
-            with a model URI.
+        registry_class : str
+            The full class name for the registry.
+        uri : str
+            The model URI.
+        name : str
+            Mode name
+        options : JavaMap
+            Options passed to the model.
         """
-        opt = model.javaOptions()
-        options = {key: opt[key] for key in opt}
-        py_class = model.pyClass()
-        logger.debug(
-            f"Code generation for model={model.toString()}, options={options}"
-            f", pyClass=${py_class}, temporary=${temporary}"
-        )
+        try:
+            registry = self.registry_map[registry_class]
+        except KeyError:
+            from rikai.internal.reflection import find_class
+
+            cls = find_class(registry_class)
+            self.registry_map[registry_class] = cls(self.spark)
```

Comment:
would it make sense to create an ABC Registry with `resolve` method?

Reply:
and we can also do an `issubclass` check here?

Reply:
i know i might be overly paranoid but these are pretty complicated internals so might be worth being more defensive / help debugging.

Reply:
Make total sense.  Will fix them